front/app/components/UI/RHFFeedback/index.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import Error, { findErrorMessage, TFieldName } from 'components/UI/Error';
+import { injectIntl } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+import {
+  Text,
+  Title,
+  Box,
+  colors,
+  Icon,
+} from '@citizenlab/cl2-component-library';
+import { scrollToElement } from 'utils/scroll';
+import CloseIconButton from '../CloseIconButton';
+import messages from './messages';
+
+type RHFFeedbackProps = {
+  successMessage?: string;
+} & InjectedIntlProps;
+
+const RHFFeedback = ({
+  successMessage,
+  intl: { formatMessage },
+}: RHFFeedbackProps) => {
+  const [successMessageIsVisible, setSuccessMessageIsShown] = useState(true);
+  const {
+    formState: { errors, isSubmitSuccessful, isSubmitted, submitCount },
+  } = useFormContext();
+
+  useEffect(() => {
+    if (submitCount > 0) {
+      scrollToElement({ id: 'rhfFeedback' });
+      setSuccessMessageIsShown(true);
+    }
+  }, [submitCount]);
+
+  const getAllErrorMessages = () => {
+    const errorMessages: Array<{ field: string; message?: string }> = [];
+
+    for (const field in errors) {
```

Comment:
More of a personal preference but could be rewritten with `Object.entries(errors).reduce(...)`

Reply:
I find this to be more readable than reduce, tbh, but indeed, it's subjective :)